prov/sockets/src/sock_av.c

```diff
@@ -527,7 +527,7 @@ int sock_av_open(struct fid_domain *domain, struct fi_av_attr *attr,
 
 		_av->table_hdr = mmap(NULL, table_sz, PROT_READ | PROT_WRITE,
 					MAP_SHARED, _av->shared_fd, 0);
-		_av->attr.map_addr = _av->table_hdr + sizeof(_av->table_hdr);
+		_av->attr.map_addr = _av->table_hdr + sizeof(struct sock_av_table_hdr);
```

Comment:
> It definitely seems that the change should be to add 1, not a sizeof(anything). That would only work if table_hdr were a void *.

:+1: on changing it to 1, also as a side note arithmetic on void pointers is a GCC extension and should probably be avoided (but you're right that it would work in the extension case since it makes `sizeof(void *) == 1`). I think the way to do that kind of addition is to first cast to an `unsigned char *`.